keras_nlp/models/opt/opt_causal_lm.py

```diff
@@ -449,3 +449,39 @@ def preprocess(x):
         if outputs.dtype == tf.string:
             return tensor_to_string_list(outputs)
         return outputs.numpy()
+
+    @classmethod
+    def create_layout_map(cls, mesh):
+        """Create a DTensor layout map for an OPTCasualLM.
```

Comment:
We could consider parameterizing the docstrings if there's a lot of copypasta going forward

Reply:
Yeah, I wasn't taking on the "base version" of this quite yet, but it would be very valid to. Probably there is very little changes that need to made here per backbone/task.